javascript/skarmbot.js

```diff
@@ -391,7 +390,76 @@ class Bot {
+                break;
+
+            case "staged":
+                if (userData.stagedImage) {
+                    e.message.channel.sendMessage(`Currently staged image submission: ${userData.stagedImage}\nTo clear this staged image, use the command \`clear\``);
+                    return;
+                } else {
+                    e.message.channel.sendMessage(`Your profile currently has no staged image to submit.`);
+                    return;
+                }
+                break;
+
+            case "clear":
+                if (userData.stagedImage) {
+                    userData.stagedImage = undefined;
+                    e.message.channel.sendMessage(`Cleared staged image.`);
+                    return;
+                } else {
+                    e.message.channel.sendMessage(`Your profile did not have a staged image to clear.`);
+                    return;
+                }
+                break;
+
+            default:
+                break;
+        }
+
+        // gets all the guilds that this user is in, and the open polls in each one
+        let openPolls = userData.getGuilds()                    // get all guilds the user is in
+            .flatMap(g => g.anonPoll.polls.filter(p => p.open)  // filter down the polls in each of those guilds to just the ones that are open to new submissions
+                .map((p) => { return { guild: g, poll: p } })); // return pairs of guild and poll to be able to print out the source guild of each poll
+
+        if (e.message.attachments.length === 1) {
+            let receivedImage = e.message.attachments[0].url;
+            userData.stagedImage = receivedImage;   // save the image to the user's data frame to be assigned to a guild with the next command
+
+            if (openPolls.length === 0) {
+                e.message.channel.sendMessage(`Received image for submission.\n\n **There are currently no open polls to submit this image to.**`);
+            } else {
+                e.message.channel.sendMessage([
+                    `Received image for submission.`,
+                    `Please send me the poll number to submit this image to:`,
+                    '```',
+                    ...openPolls.map((gp, idx) => `[${idx + 1}] -> [Server: ${gp.guild.getName()}] Poll: ${gp.poll.name}`),
+                    '```',
+                ].join("\n"));
+            }
+
+            return;
+        }
+
+
+        let idx = (Number(msg) - 1);
+        if(idx in openPolls){ // validity check
+            let {guild, poll} = openPolls[idx];
+            poll.submit(userData.id, userData.stagedImage);
+            e.message.channel.sendMessage(`Submitted the image ${userData.stagedImage} to the poll ${poll.name} (${guild.getName()})`);
+            userData.stagedImage = undefined; // drop the staged image from the user once it is pushed to the poll
+            return;
+        }
 
+        e.message.channel.sendMessage("Unexpected direct message recieved.");
     }
```

Comment:
We probably want all of this to be migrated out to the commands folder in the future, along with some tag to differentiate guild-compatible commands and DM-compatible commands